### PR TITLE
Added proper way to share stories

### PR DESCRIPTION
**Pull request by doba16**
### Description

Added a proper way to share stories.

### CODE REVIEW

File: taletime/android/app/build.gradle [hunk 1]
```diff
@@ -36,7 +36,7 @@ if (flutterVersionName == null) {
 
 
 android {
-    compileSdkVersion 33
+    compileSdkVersion 34
```
Comment (SamTV12345) [inline taletime/android/app/build.gradle:39]: Ah nice. That will solve the other pr

File: taletime/lib/listener/screens/listener_homepage.dart [hunk 2]
```diff
@@ -223,7 +223,7 @@ class _ListenerHomePageState extends State<ListenerHomePage> {
                 Icons.menu,
                 size: 33, color: kPrimaryColor, //kPrimaryColor
```
Comment (SamTV12345) [inline taletime/lib/listener/screens/listener_homepage.dart:224]: What about the comment? Maybe break the lines here

File: taletime/lib/login and registration/utils/authentification_util.dart [hunk 3]
```diff
@@ -1,13 +1,14 @@
 import "package:cloud_firestore/cloud_firestore.dart";
 import "package:firebase_auth/firebase_auth.dart";
 import "package:flutter/material.dart";
+import "package:taletime/common%20utils/constants.dart";
 import "package:taletime/common%20utils/tale_time_logger.dart";
 import "package:taletime/common/models/tale_time_user.dart";
 import "package:taletime/login%20and%20registration/screens/login.dart";
+import "package:taletime/login%20and%20registration/utils/error_util.dart";
```
Comment (SamTV12345) [inline taletime/lib/login and registration/utils/authentification_util.dart:8]: Maybe we should remove the white spaces here from the package name

File: taletime/lib/main.dart [hunk 4]
```diff
@@ -81,9 +86,23 @@ class Providers extends StatelessWidget {
...
     final languageProvider = Provider.of<LocaleProvider>(context);
     final themeProvider = Provider.of<ThemeProvider>(context);
-    return MaterialApp(
+
+    return MaterialApp.router(
+      routerConfig: GoRouter(
+        routes: [
+          GoRoute(
+            path: "/",
+            builder: (_, __) => HomePage(),
```
Comment (SamTV12345) [inline taletime/lib/main.dart:95]: Can't we just remove the _ and just have ()=>?

File: taletime/lib/main.dart [hunk 5]
```diff
@@ -81,9 +86,23 @@ class Providers extends StatelessWidget {
...
+            routes: [
+              GoRoute(
+                path: "shared",
+                builder: (_, __) => SharedStory(),
```
Comment (SamTV12345) [inline taletime/lib/main.dart:99]: Same here

File: taletime/lib/player/widgets/player_controls.dart [hunk 6]
```diff
@@ -1,11 +1,28 @@
...
+
+    if (mediaItem == null) {
+      return;
+    }
+
+    final title = mediaItem.title;
+    final author = mediaItem.artist;
+    final id = mediaItem.id;
+
+    final sharedText = "$title ${AppLocalizations.of(context)!.by} $author - https://taletime-2022.web.app/shared?storyId=$id";
```
Comment (SamTV12345) [inline taletime/lib/player/widgets/player_controls.dart:22]: Maybe the base url should be put in an env file or something so once we scale we don't have to swap all links by https://taletime.app/shared?storyId=$id":)

File: taletime/lib/share/screens/shared_story.dart [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+        ],
+      );
+    }
+
+    // Test if logged in
+    if (Provider.of<UserState>(context).user == null) {
+      return renderWidget(
+        profile: Text(AppLocalizations.of(context)!.notLoggedIn),
+        btnClick: story == null
+            ? null
```
Comment (SamTV12345) [inline taletime/lib/share/screens/shared_story.dart:125]: Don't know what is more readable. Simply story or null

**Review by SamTV12345 — CHANGES_REQUESTED** [review]

File: taletime/lib/main.dart (hunk 4 above)
Comment (doba16) [reply taletime/lib/main.dart:95]: Nope, does not work.

File: taletime/lib/main.dart (hunk 5 above)
Comment (doba16) [reply taletime/lib/main.dart:99]: Nope.

File: taletime/lib/share/screens/shared_story.dart (hunk 7 above)
Comment (doba16) [reply taletime/lib/share/screens/shared_story.dart:125]: This would lead to incompatible types.

**Review by SamTV12345 — APPROVED** [review]

**Pull request merged**